docs/source/api.rst

```diff
@@ -173,6 +173,7 @@ Other Algorithm Functions
    retworkx.core_number
    retworkx.graph_greedy_color
    retworkx.metric_closure
+   retworkx.all_pairs_all_simple_paths
```

Comment:
I'm thinking if it's more appropriate to move this (and `retworkx.all_simple_paths`) under `Connectivity` section or even make another separate section.

Reply:
Sure moving it under connectivity makes sense, I just put it here because that's where `all_simple_paths` was. I'll move this now.

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/577/commits/edda11dc104ccefd0005624ddde82eb89d5edb59